README.md

```diff
@@ -0,0 +1,101 @@
+
+The TypeSpec MCP Server implements the [MCP specification](https://modelcontextprotocol.io) to enable you to quickly build out your own MCP extensions powered by [TypeSpec](https://typespec.io/)
+
+> Please note that this project is in Public Preview and implementation may significantly change prior to our General Availability.
+
+## Overview
+
+### What packages are in this repository?
+
+Currently this repo contains the following packages:
+
+- [mcp-server-typespec](https://github.com/bterlson/typespec-mcp/tree/main/packages/mcp-server-typespec) - an MCP server which assists in developing MCP extensions using TypeSpec.
+- [typespec-mcp](https://github.com/bterlson/typespec-mcp/tree/main/packages/typespec-mcp) - a TypeSpec library that provides a `tool` decorator.
+- [typespec-mcp-server-js](https://github.com/bterlson/typespec-mcp/tree/main/packages/typespec-mcp-server-js) - A TypeSpec emitter that generates a server implementation in JavaScript for an MCP tool.
+- [typespec-mcp-http-server-js](https://github.com/bterlson/typespec-mcp/tree/main/packages/typespec-mcp-http-server-js) - A TypeSpec emitter that generates a server implementation in JavaScript for calling a remote REST API.
+
+## Currently Supported Tools
+
+The Azure MCP Server provides tools for interacting with the following Azure services:
+
+- `learnTypeSpec` - Initializes the model with information about how to understand and write TypeSpec.
+- `init` - Scaffolds out a new project in the current working directory with example tool implementation.
+- `compile` - Runs tsp compile to generate emitter assets
+- `build` - Executes `npm run build` in the current project.
+
+
+## Getting Started
+
+The TypeSpec MCP Server requires Node.js to install and run the server. If you don't have it installed, follow the instructions [here](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+
+### VS Code + GitHub Copilot
+
+The TypeSpec MCP Server provides assistance with authoring TypeSpec to generate MCP resource implementations. It can be used alone or with the [GitHub Copilot for Azure extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azure-github-copilot) in VS Code.
+
+### Prerequisites
+1. Install either the stable or Insiders release of VS Code:
+   * [Stable release](https://code.visualstudio.com/download)
+   * [Insiders release](https://code.visualstudio.com/insiders)
+2. Install the [GitHub Copilot](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot) and [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) extensions
+3. Install [Node.js](https://nodejs.org/en/download) 22 or later
```

Comment:
Personally I think we should support any maintained Node LTS which is 20+ while it is still maintained. What features do we use that is 22+ other than import.meta.dirname (which we can polyfill relatively easily)?

Reply:
agreed, will push an update